Plugins/NativeServerPlugin/src/directx_buffer_capturer.cpp

```diff
@@ -0,0 +1,258 @@
+#include "pch.h"
+
+#include "directx_buffer_capturer.h"
+#include "plugindefs.h"
+
+#include "webrtc/modules/video_coding/codecs/h264/h264_encoder_impl.h"
+
+using namespace Microsoft::WRL;
+using namespace StreamingToolkit;
+
+DirectXBufferCapturer::DirectXBufferCapturer(ID3D11Device* d3d_device) :
+	d3d_device_(d3d_device)
+{
+}
+
+void DirectXBufferCapturer::Initialize(bool headless, int width, int height)
+{
+	// Gets the device context.
+	d3d_device_->GetImmediateContext(&d3d_context_);
+
+#ifdef MULTITHREAD_PROTECTION
```

Comment:
do we define this in the right places (mostly a note to self for future review of this PR, will close out when i've verified)

Reply:
i don't see this anywhere in `Plugins/NativeServerPlugin/StreamingNativeServerPlugin.vcxproj` is the expectation that one of our builds will have this enabled, or just that a build could be produced that should behave this way? If it's the latter, let's document this directive somewhere

Reply:
It's defined in *plugindefs.h*.